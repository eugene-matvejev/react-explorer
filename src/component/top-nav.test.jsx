import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import _TopNav from './top-nav';

const TopNav = (props) =>
    <MemoryRouter>
        <_TopNav {...props} />
    </MemoryRouter>;

describe('<TopNav />', () => {
    const props = {
        label: '{{label}}',
    };

    describe('render', () => {
        it('with default/required props', () => {
            const { container, debug } = render(<TopNav {...props} />);

            expect(container.querySelector('[data-cy="topnav-main"]')).toBeInTheDocument();
            expect(container.querySelector('[data-cy="topnav-github"]')).toBeInTheDocument();
        });
    });
});
