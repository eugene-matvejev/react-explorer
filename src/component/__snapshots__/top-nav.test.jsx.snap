// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TopNav/> render with default/require d props 1`] = `
<nav
  className="topnav "
  label="{{label}}"
>
  <img
    alt="logo"
    className="topnav__logo"
    src="/assets/img/placeholder.png"
  />
  <Link
    target="_blank"
    to="//github.com/eugene-matvejev/react-explorer"
  >
    <img
      alt=" GitHub logo"
      className="topnav__github-logo"
      src="/assets/img/github-logo.png"
    />
  </Link>
</nav>
`;

exports[`<TopNav/> render with optional props [::className] as "{{className}}" 1`] = `
<nav
  className="topnav {{className}}"
  label="{{label}}"
>
  <img
    alt="logo"
    className="topnav__logo"
    src="/assets/img/placeholder.png"
  />
  <Link
    target="_blank"
    to="//github.com/eugene-matvejev/react-explorer"
  >
    <img
      alt=" GitHub logo"
      className="topnav__github-logo"
      src="/assets/img/github-logo.png"
    />
  </Link>
</nav>
`;

exports[`<TopNav/> render with optional props [::data-cy] as "{{data-cy}}" 1`] = `
<nav
  className="topnav "
  data-cy="{{data-cy}}"
  label="{{label}}"
>
  <img
    alt="logo"
    className="topnav__logo"
    src="/assets/img/placeholder.png"
  />
  <Link
    target="_blank"
    to="//github.com/eugene-matvejev/react-explorer"
  >
    <img
      alt=" GitHub logo"
      className="topnav__github-logo"
      src="/assets/img/github-logo.png"
    />
  </Link>
</nav>
`;
